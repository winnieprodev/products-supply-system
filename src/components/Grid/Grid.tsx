import { AnimatedGroup } from "components";
import React from "react";

import * as Styled from "./grid.styled";

interface OwnProps {
  itemsInRow?: number;
  centered?: boolean;
}

const Grid: React.FC<React.PropsWithChildren<OwnProps>> = ({
  itemsInRow = 5,
  centered = false,
  children,
}) => (
  <Styled.Grid $itemsInRow={itemsInRow} $centered={centered}>
    <AnimatedGroup>{children}</AnimatedGroup>
  </Styled.Grid>
);

export default Grid;
