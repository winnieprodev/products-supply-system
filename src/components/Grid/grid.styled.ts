import styled, { css } from "styled-components";

interface GridProps {
  $itemsInRow: number;
  $centered: boolean;
}

export const Grid = styled.div<GridProps>`
  margin: 30px 0;
  display: flex;
  flex-wrap: wrap;
  ${({ $centered }) =>
    $centered &&
    css`
      justify-content: center;
    `}

  /* we must skip AnimatedGroup's body */
  & > * > * {
    width: calc(${({ $itemsInRow }) => Math.round(100 / $itemsInRow)}% - 24px);
    margin: 12px;
  }
`;
