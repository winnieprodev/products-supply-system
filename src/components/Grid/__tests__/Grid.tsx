import { render } from "@testing-library/react";

import Grid from "../Grid";

describe("Grid", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for basic use", () => {
      const { container } = render(
        <Grid>
          <div>item</div>
          <div>item</div>
        </Grid>,
      );
      expect(container).toMatchSnapshot();
    });
    test("it matches the snapshot with props", () => {
      const { container } = render(
        <Grid centered>
          <div>item</div>
          <div>item</div>
        </Grid>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
