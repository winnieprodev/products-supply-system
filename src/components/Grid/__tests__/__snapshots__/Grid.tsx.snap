// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snapshot tests it matches the snapshot for basic use 1`] = `
.c1 {
  display: contents;
}

.c1 > .animated {
  opacity: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c0 {
  margin: 30px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 > * > * {
  width: calc(20% - 24px);
  margin: 12px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="animated"
      >
        item
      </div>
      <div
        class="animated"
      >
        item
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid Snapshot tests it matches the snapshot with props 1`] = `
.c1 {
  display: contents;
}

.c1 > .animated {
  opacity: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c0 {
  margin: 30px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > * > * {
  width: calc(20% - 24px);
  margin: 12px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="animated"
      >
        item
      </div>
      <div
        class="animated"
      >
        item
      </div>
    </div>
  </div>
</div>
`;
